src/classes/Trades.ts

```diff
@@ -439,31 +439,48 @@ export default class Trades {
         });
     }
 
-    acceptConfirmation(offer: TradeOfferManager.TradeOffer): Promise<void> {
+    acceptConfirmation(offer: TradeOffer, attempts = 0): Promise<void> {
+        attempts++;
+        if (
+            offer.state === TradeOfferManager.ETradeOfferState['Active'] ||
+            offer.state === TradeOfferManager.ETradeOfferState['CreatedNeedsConfirmation']
+        ) {
+            return this.acceptConfirmationPromise(offer, attempts).catch(async (err: Error) => {
+                if (attempts > 3) {
+                    throw err;
+                }
+
+                await promiseDelay(20 * 1000);
+
+                return this.acceptConfirmation(offer, attempts);
+            });
+        }
+    }
+
+    acceptConfirmationPromise(offer: TradeOffer, attempts = 1): Promise<void> {
         return new Promise((resolve, reject) => {
-            log.debug('Accepting mobile confirmation...', {
-                offerId: offer.id
+            log.debug(`Accepting mobile confirmation...`, {
+                offerId: offer.id,
+                attempts: attempts
             });
+            offer.data('attempts', attempts);
 
             const start = dayjs().valueOf();
             offer.data('actedOnConfirmation', true);
             offer.data('actedOnConfirmationTimestamp', start);
 
             this.bot.community.acceptConfirmationForObject(this.bot.options.steamIdentitySecret, offer.id, err => {
-                const confirmationTime = dayjs().valueOf() - start;
-                offer.data('confirmationTime', confirmationTime);
-
                 if (err) {
-                    log.debug(`Error while trying to accept mobile confirmation on offer #${offer.id}: `, err);
-                    return reject(err);
+                    reject(err);
+                    return;
```

Comment:
if you reject there is no return because the last statement is a function call, effectively a void

Reply:
ah right. All return no need for promises right?

Reply:
oh wait nvm

Reply:
async should return Promises but when you "promisify" ie make a new promise with callbacks via a new Promise((resolve, reject) => {} those only need resolve and reject calls. You would return that new Promise to the callee though.

Reply:
understood.